src/steps/create-template-only-components.ts

```diff
@@ -8,35 +8,40 @@ import type { Context, Options } from '../types/index.js';
 import { blueprintsRoot } from '../utils/blueprints.js';
 import { getComponentFilePath, transformEntityName } from '../utils/files.js';
 
-function createBackingClass(entityName: string, options: Options): void {
-  const filePath = getComponentFilePath(options)(entityName);
+const blueprintFile = readFileSync(
+  join(blueprintsRoot, 'ember-cli/template-only-component.ts'),
+  'utf8',
+);
 
-  const blueprintFile = readFileSync(
-    join(blueprintsRoot, 'ember-cli/template-only-component.ts'),
-    'utf8',
-  );
+function createBackingClass(entityName: string, options: Options): string {
+  const entity = transformEntityName(entityName);
 
   const file = processTemplate(blueprintFile, {
-    entity: transformEntityName(entityName),
+    entity,
     options,
   });
 
-  createFiles(new Map([[filePath, file]]), options);
+  return file;
 }
 
 export function createTemplateOnlyComponents(
   context: Context,
   options: Options,
 ): void {
+  const fileMap = new Map<string, string>();
+
   for (const [entityName, extensions] of context.entities) {
     const hasBackingClass = extensions.has('.ts');
 
     if (hasBackingClass) {
       continue;
     }
 
-    createBackingClass(entityName, options);
+    const filePath = getComponentFilePath(options)(entityName);
+    const file = createBackingClass(entityName, options);
 
-    extensions.add('.ts');
```

Comment:
**bugfix**

I found out, running the tests for `create-template-only-components`, which would mutate the shared `context` in tests, would cause the tests for the previous step, `analyze-project`, to fail.

Luckily, it turned out that the line `extensions.add('.ts')` wasn't needed for the subsequent steps `create-signatures` and `create-registries`.